autoscheduler/scraper/management/commands/scrape_courses.py

```diff
@@ -329,6 +330,31 @@ def save_models(instructors: List[Instructor], sections: List[Section], # pylint
 
     print(f"Saved all in {elapsed_time:.2f} seconds")
 
+def save_terms(term, terms, options):
+    """ Creates terms objects to save """
+
+    start = time.time()
+    now = timezone.now() # use timezone.now() so Django doesn't complain about naive times
+    with transaction.atomic():
+        # Bulk upsert
+        if term: # If a specific term was provided, we're only updating that specific one
+            term_models = [Term(code=term, last_updated=now)]
+            queryset = Term.objects.filter(code=term)
+        else: # otherwise, we're scraping multiple terms
+            term_models = [Term(code=term, last_updated=now) for term in terms]
+
+            if options['year'] or options['recent']:
+                queryset = Term.objects.filter(code__in=terms)
+            else: # If no options are provided, then we're scraping all terms
+                queryset = Term.objects.all()
+
+        queryset.delete()
+        Term.objects.bulk_create(term_models)
+
+        term_len = len(terms) if terms else 1
```

Comment:
This code is kinda ugly and would be a lot nicer by just providing `terms` like I describe below

Reply:
I'm just gonna do this now - we probably wouldn't get around to it otherwise and it's really bugging me now that you pointed it now